packages/vm/src/cache.rs

```diff
@@ -88,7 +88,7 @@ where
     pub fn save_wasm(&mut self, wasm: &[u8]) -> VmResult<Checksum> {
         check_wasm(wasm, &self.supported_features)?;
         let checksum = save_wasm_to_disk(&self.wasm_path, wasm)?;
-        let module = compile(wasm, None)?;
+        let module = compile_only(wasm)?;
```

Comment:
If we don't set the memory limit here, where do we set it?

`Instance::from_module()` doesn't have a spot to do so

Reply:
The memory limit is set using `make_runtime_store` when loading it from cache